src/erlcloud_ddb2.erl

```diff
@@ -61,12 +61,17 @@
 %% contain typed attribute values so that they may be correctly passed
 %% to subsequent calls.
 %%
-%% DynamoDB errors are return in the form `{error, {ErrorCode,
-%% Message}}' where `ErrorCode' and 'Message' are both binary
+%% DynamoDB errors for most cases are return in the form
+%% `{error, {ErrorCode, Message}}' where `ErrorCode' and `Message' are both binary
 %% strings. List of error codes:
 %% [http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ErrorHandling.html]. So
 %% to handle conditional check failures, match `{error,
 %% {<<"ConditionalCheckFailedException">>, _}}'.
+%% Note that for `TransactionCanceledException' DynamoDB error type, it is returned
+%% in the form `{error, {<<"TransactionCanceledException">>, {Message,
+%% CancellationReasons}}}' where `Message' is a binary string and `CancellationReasons'
+%% is a proplist in the form `[{Code, Message}]` where `Code' is the status code of the result
```

Comment:
I think we should call this an "ordered list" or something. And use `ErrorCode` rather than `Code` for consistency.

Reply:
AWS calls it `Code` since it is not always an error.

Reply:
`ReasonCode` maybe?

Reply:
Good point. Never mind, leave it as is.